docs/articles/game-engines/article_01.md

```diff
@@ -0,0 +1,189 @@
+consider when optimizing performance and while virtual lookup tables may not be
+the first culprit for performance issues, it is among the few that crop up time
+and time again.
+
+I personally don't think that you should change the way you like to program for
+the sake of someone's opinions about what the "right way" to program is. I believe
+that every developer has a choice to write code the way they like to write it.
+However, I would implore you to focus more about writing code that completes the
+task at hand rather than writing code you think you will need in the future.
+
+### Over-Optimization and You
+
+This brings us to the next point: over-optimization. We want to write code that is
+both fast and easy to write, but rarely do these two things happen together. Throughout
+these articles, I will demonstrate code which is the easiest to write. Such code will
+probably stay the way it is for a significant amount of time, other times we will
+immediately refactor it to account for performance. It's easy to get carried away
+with your code and so the first key advice I have for you is to slow down.
+
+One of the best ways to approach any given problem, be it programming or in life,
+is to look at with a "first principles" perspective. How would you solve this problem
+if you were told to do this using your gut instinct? Your solution need not be elegant,
+fast, or thought out. You might be surprised that simple solutions are often some of
+the best. Our goal is not to write the best code around, but to write code that gets
+us to step two, and then to step three, and so on.
+
+When performance problems arise, they stick out. That's when you optimize.
+
+### Defining Your Code Style
+
+I write code in a very particular way. I liken it more to C than to C++. For object
+composition, I use structs to handle my data arrangements. Sometimes, I use static
+globals. When I write functions, I write long bodies of code that could be reasonably
+decomposed into smaller functions. You could say that I break all the
+conventions of what "clean code" looks like.
+
+There are no rules to software development, my friends. If there were, my code
+wouldn't compile. My second bit of key advice to you is to find a code style that
+works best for you and stick with it. Do not change how you do things mid way through
+your project and refactor everything just to meet your new standards. If you absolutely
+must, don't refactor, just switch. I don't know of any code base that doesn't look
+like a nightmare after line 10,000. That is just what scale looks like.
+
+## Up Next: Setup and Win32 Basics
+
+In the next article we will set up our development environment using CMake and
+look to build out our application with the Win32 API. We will explore Window's
+application main entry point, defining some primitive data types, and launch our
+window.
+
+[>> Article 02: Setup and Win32 Basics](./article_02.md)
```

Comment:
To keep in line with the style of other articles, structure "next" and "previous" links as so:

`Previous: [link] (if applicable)`
`Next: [link] (if applicable)`

Write `(Coming Soon!) after the link text if the page it refers to does not exist yet.